source/data-modeling/associations.txt

```diff
@@ -0,0 +1,504 @@
+   :start-after: # start-association-query
+   :end-before: # end-association-query
+
+Embedded Associations
+---------------------
+
+You can use embedded associations to store different types of documents in the
+same collection. {+odm+} supports embedded associations with the following 
+macros:
+
+- ``embeds_one``
+- ``embeds_many``
+- ``embedded_in``
+- ``recursively_embeds_one``
+- ``recursively_embeds_many``
+
+The following sections describe how to use these association types.
+
+Embeds One
+~~~~~~~~~~
+
+To specify that a class model contains an embedded document of a different
+class type, use the ``embeds_one`` macro in the parent class and the ``embedded_in``
+macro in the embedded class. The following example creates a ``Band`` class with
+an embedded ``Label`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-one
+   :end-before: # end-embed-one
+
+{+odm+} stores documents embedded with the ``embeds_one`` macro in the
+parent document as a field with the same name as the embedded class. The
+preceding ``Label`` documents are stored in the ``Band`` document as shown in
+the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-one-stored
+   :end-before: # end-embed-one-stored
+
+You can store the embedded document with a different name by using the
+``store_as`` option, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-store-as
+   :end-before: # end-embed-store-as
+
+Embeds Many
+~~~~~~~~~~~
+
+To specify that a class model contains multiple embedded documents of a
+different class type, use the ``embeds_many`` macro in the parent class and the
+``embedded_in`` macro in the embedded class. The following example creates a
+``Band`` class with multiple embedded ``Album`` type documents:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many
+   :end-before: # end-embed-many
+
+{+odm+} stores documents embedded with the ``embeds_many`` macro in the
+parent document as an array field with the same name as the embedded class. The
+preceding ``Album`` documents are stored in the ``Band`` document as shown in
+the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many-stored
+   :end-before: # end-embed-many-stored
+
+You can store the embedded document with a different name by using the
+``store_as`` option, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many-store-as
+   :end-before: # end-embed-many-store-as
+
+Recursive Embedding
+~~~~~~~~~~~~~~~~~~~
+
+You can recursively embed one or more documents of the same type into a parent
```

Comment:
S: can you briefly describe what it means to recursively embed a document? Unless you're assuming the reader is familiar with the concept

Reply:
I think this whole section is essentially an explanation of what "recursive embedding" is, so I removed "recursively embed" from the first sentence to make it seem less like we're introducing a new term here.